src/SURFnet/OATHBundle/Services/UserStorage/Encryption/Openssl.php

```diff
@@ -1,47 +1,54 @@
-<?php 
+<?php
 
 namespace SURFnet\OATHBundle\Services\UserStorage\Encryption;
 
 /**
  * Class for encrypting/decrypting the user secret with openssl.
- * 
+ *
  * @author peter
  */
 class Openssl implements UserEncryptionInterface
 {
-    private $_method;
-    private $_key;
+    /**
+     * @var string
+     */
+    private $method;
+    /**
+     * @var string
+     */
+    private $key;
 
     /**
      * Construct an encryption instance.
      *
-     * @param $config The configuration that a specific configuration class may use.
+     * @param $config array The configuration that a specific configuration class may use.
+     * @throws \Exception
      */
     public function __construct($config)
     {
-        $this->_method = $config['method'];
+        $this->method = $config['method'];
```

Comment:
This configuration solution feels really fragile. This condition could have been verified during container compile time. This would prevent the application to run with a faulty configuration.